File: otherLanguageHelpers/java-nodered-helpers/testerapp/src/main/java/d/nrtest/genpurpbenchmark/Benchmark.java
package d.nrtest.genpurpbenchmark;

import d.nrtest.nrcommon.PerformanceMeter;

import java.util.Random;

/**
 * @author Domen Mori 20. 10. 2018.
 */
public class Benchmark {

    private static final int ARRAY_LENGHT = 30000000;
    private static int writer = 0;

    private static int incrementWriter(){
        int r = writer++;
//        writer = (writer + 1) % ARRAY_LENGHT;
        if (writer >= ARRAY_LENGHT){
            writer = 0;
        }
        return r;
    }

    public static void main(String[] args) {

        Random rand = new Random();
        PerformanceMeter meter = new PerformanceMeter(2000000);

        long []first = new long[ARRAY_LENGHT];
        long []second = new long[ARRAY_LENGHT];

        for(int i = 0; i<ARRAY_LENGHT; ++i){
            first[i] = rand.nextLong();
            second[i] = rand.nextLong();
        }

        long [] results = new long[ARRAY_LENGHT*2];

        while (true){
            results[incrementWriter()] = first[rand.nextInt(ARRAY_LENGHT)] + second[rand.nextInt(ARRAY_LENGHT)];
            results[incrementWriter()] = (long)((double)first[rand.nextInt(ARRAY_LENGHT)] / (double)second[rand.nextInt(ARRAY_LENGHT)]);
            results[incrementWriter()] = first[rand.nextInt(ARRAY_LENGHT)] % second[rand.nextInt(ARRAY_LENGHT)];
            meter.notifyEvent();
        }
    }
}

File: otherLanguageHelpers/go/mqttRunner/cmd/BenchmarkThingLegacy/benchmark.go
package main

import (
	"mqttRunner/pkg/utils/performancemeter"
	"math/rand"
)

const ARRAY_LENGHT = 30000000
var writer int32 = 0

func incrementWriter() int32 {
	r := writer
	writer++
	if writer >= ARRAY_LENGHT{
		writer = 0;
	}
	return r
}

func main() {
	var first [ARRAY_LENGHT]int64
	var second [ARRAY_LENGHT]int64

	for i:=0; i<ARRAY_LENGHT; i++ {
		first[i] = rand.Int63()
		second[i] = rand.Int63()
	}

	meter := performancemeter.Initialize(3699055)

	var results [ARRAY_LENGHT*2]int64

	for {
		results[incrementWriter()] = first[rand.Int31n (ARRAY_LENGHT)] + second[rand.Int31n(ARRAY_LENGHT)];
		results[incrementWriter()] = int64(float64(first[rand.Int31n(ARRAY_LENGHT)]) / float64(second[rand.Int31n(ARRAY_LENGHT)]));
		results[incrementWriter()] = first[rand.Int31n(ARRAY_LENGHT)] % second[rand.Int31n(ARRAY_LENGHT)];
		performancemeter.NotifyEvent(meter)
	}

}